packages/autocomplete-js/src/__tests__/components.test.tsx

```diff
@@ -0,0 +1,430 @@
+/** @jsx h */
+import { Hit } from '@algolia/client-search';
+import { fireEvent, waitFor } from '@testing-library/dom';
+import { h } from 'preact';
+
+import { createSource } from '../../../../test/utils';
+import { autocomplete } from '../autocomplete';
+
+import products from './fixtures/products.json';
+
+type ProductRecord = {
+  brand: string;
+  categories: string[];
+  description: string;
+  free_shipping: boolean;
+  hierarchicalCategories: {
+    lvl0: string;
+    lvl1?: string;
+    lvl2?: string;
+    lvl3?: string;
+    lvl4?: string;
+    lvl5?: string;
+    lvl6?: string;
+  };
+  image: string;
+  name: string;
+  popularity: number;
+  price: number;
+  prince_range: string;
+  rating: number;
+  type: string;
+};
+type ProductHit = Hit<ProductRecord>;
+
+const productHits = products.results[0].hits;
+
+describe('components', () => {
+  afterEach(() => {
+    document.body.innerHTML = '';
+  });
+
+  test('provides Highlight component', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<ProductHit>({
+      container,
+      panelContainer,
+      getSources() {
+        return [
+          {
+            ...createSource({
+              getItems() {
+                return productHits;
+              },
+            }),
+            templates: {
+              item({ item, components }) {
+                return <components.Highlight hit={item} attribute="name" />;
```

Comment:
I'm slightly surprised it ships with these four components already, won't most implementations only need one specific one?

Reply:
You mean that you would only expect `Highlight` to be included, and not `Snippet`, etc.?

I wanted to provide the tools so that users don't have to think about importing those. I'll check the bundle size impact of that decision to see if it's worth including all.

Reply:
good point! I was thinking mostly about bundlesize, but if it's small enough, it can stay like this

Reply:
Here are the results:

- Without components: 13.83KB
- With all components: 14.63KB
- With just `Highlight`: 14.18KB

I'm still thinking if it's worth it. Any opinions yet?

Reply:
yes, 1kb for all the components makes sense. At a later point we can see if there's a way to deduplicate some code in them or externalising :)

Reply:
I had the same concerns initially but 1 extra KB is definitely a no-brainer compared to the DX it brings.